### PR TITLE
MGDOBR-595: fix openapi generation for enums

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-595](https://issues.redhat.com/browse/MGDOBR-595)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

**Review by kelvah — APPROVED** [review]
LGTM. Thank you @r00ta :+1:

File: infra/pom.xml [hunk 1]
```diff
@@ -55,6 +55,10 @@
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-resteasy-qute</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-openapi</artifactId>
+    </dependency>
```
Comment (danielezonca) [inline infra/pom.xml:61]: Is this necessary? I don't see any additional/new usage in the PR

Comment (r00ta) [reply infra/pom.xml]: removed

**Review by danielezonca — APPROVED** [review]
LGTM 👍

**Review by jomarko — COMMENTED** [review]
Hello team, I have a question about this enum change, because also sonar reports for some enums we do not follow naming convention 

e.g. https://sonarcloud.io/project/issues?resolved=false&severities=CRITICAL&types=CODE_SMELL&id=5733d9e2be6485d52ffa08870cabdee0_sandbox&open=AX_Zl5Sho54ijqCYTPng

So my question is if we should somehow revisit all enums in sandbox project to:
- follow serializing introduced in this PR
- follow standard java convention - items uppercase CAR instead of Car

**Comment by danielezonca** [comment]
@jomarko 
I agree that we should follow the approach of this PR for all Enums that we serialise so that we can follow Java guidelines and have a different value during the serialisation

**Comment by jomarko** [comment]
@danielezonca how can I check what enum is serialized and what is used just internally please? Is that just if some `@Json` annotation is present?

**Comment by danielezonca** [comment]
> @danielezonca how can I check what enum is serialized and what is used just internally please? Is that just if some `@Json` annotation is present?

I don't think there is a simple way to do this check. I expect DTO to be well isolated in `infra` / `models` packages/modules but this is not enough to guarantee it. I would start with Enum where we are not following Java guidelines
The Json annotation is not mandatory as far as I remember

**Comment by r00ta** [comment]
@jomarko shall we merge this?

**Review by jomarko — COMMENTED** [review]
Yes we can, I reported https://issues.redhat.com/browse/MGDOBR-639 for me to work on this later.

**Review by jomarko — APPROVED** [review]

**Comment by r00ta** [comment]
\rebase

**Comment by r00ta** [comment]
> @r00ta @kelvah is desperate for this fix.. could you please let him know when this is merged :+1: 

As sono as CI Is green..sure :)

**Pull request merged**